src/Identity/UI/src/IdentityBuilderUIExtensions.cs

```diff
@@ -131,5 +125,31 @@ private static void AddRelatedParts(IdentityBuilder builder, UIFramework framewo
                     }
                 });
         }
+
+        private static Assembly GetApplicationAssembly(IdentityBuilder builder)
+        {
+            // Whis is the same logic that MVC follows to find the application assembly.
+            var environment = builder.Services.Where(d => d.ServiceType == typeof(IWebHostEnvironment)).ToArray();
+            var applicationName = ((IWebHostEnvironment)environment.LastOrDefault()?.ImplementationInstance)
+                .ApplicationName;
+
+            var appAssembly = Assembly.Load(applicationName);
+            return appAssembly;
+        }
+
+        private static bool TryResolveUIFramework(Assembly assembly, out UIFramework uiFramework)
+        {
+            var metadata = assembly.GetCustomAttributes<UIFrameworkAttribute>()
+                .SingleOrDefault()?.UIFramework; // Bootstrap4 is the default
+            if (metadata == null)
+            {
+                uiFramework = default;
+                return false;
+            }
+
+            // If we find the metadata there must be a valid framework here.
+            uiFramework = Enum.Parse<UIFramework>(metadata, ignoreCase: true);
```

Comment:
Could this throw a meaningful error if the value is unsupported?

Reply:
done